### PR TITLE
✨ Add support for NSPhotoLibraryUsageDescription

### PR DESCRIPTION
**Pull request by FransRosencrantz**
* Fix QT_IOS_ITS_ENCRYPTION_KEYS should be true if
QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE is set?

### CODE REVIEW

File: AddQtIosApp.cmake [hunk 1]
```diff
@@ -155,7 +158,7 @@ function(add_qt_ios_app TARGET)
     endif()
 
     # Allow user to override QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE from cache/command line
-    if(NOT QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE)
+    if(QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE)
```
Comment (OlivierLDff) [inline AddQtIosApp.cmake:161]: I believe `if(NOT QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE)` is correct.
The goal is too be able to develop without it, don't hardcode it in CMakeLists.
Then in Ci for example you can add `cmake -DQT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE=123345678765434`
I don't have such code, so i don't know how it is expected to look like

File: AddQtIosApp.cmake [hunk 2]
```diff
@@ -99,6 +100,7 @@ function(add_qt_ios_app TARGET)
         CATALOG_APPICON
         CATALOG_LAUNCHIMAGE
         DISTRIBUTION_METHOD
+        PHOTO_LIBRARY_USAGE_DES
```
Comment (OlivierLDff) [inline AddQtIosApp.cmake:103]: Can you use long variable name : `PHOTO_LIBRARY_USAGE_DESCRIPTION`.
I know it's long but when going back to it after month it's quite nice to understand description and not spending understanding what DES is. (Maybe Des is very common in english, but i never saw it as french person). Desc is more common to me.
What do you think?

**Review by OlivierLDff — CHANGES_REQUESTED** [review]
Is it also possible to add a dedicated section in the Readme?
And with a screenshot of what it is doing in XCode (if this is relevant)

File: AddQtIosApp.cmake (hunk 1 above)
Comment (FransRosencrantz) [reply AddQtIosApp.cmake:161]: My mistake, you're right! Must have had QT_IOS_ITS_ENCRYPTION_EXPORT_COMPLIANCE_CODE cached which made me confused.

File: AddQtIosApp.cmake (hunk 2 above)
Comment (FransRosencrantz) [reply AddQtIosApp.cmake:103]: I agree let's go for `PHOTO_LIBRARY_USAGE_DESCRIPTION`

File: AddQtIosApp.cmake (hunk 1 above)
Comment (OlivierLDff) [reply AddQtIosApp.cmake:161]: Yes, as you see quite lots of variable are intended to be used that way.
Apple make it quite hard for dev ahah
image

**Comment by FransRosencrantz** [comment]
@OlivierLDff Hmm I just saw that there are many different Cocoa Keys https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html
In our project, we just need one of these keys  "NSPhotoLibraryUsageDescription" but maybe you don't want to have just one of these variables in the repo? In that case, do you have a better idea to add it? Combining the custom list with the generated? maybe?

**Comment by OlivierLDff** [comment]
The idea is for you to copy the `Info.plist.in` inside your project and provide it via `CUSTOM_PLIST`.
That way you can add all the keys you need in it.
It can be interesting for QtIosCMake to already come with couple of common key. I believe that accessing photo library is standard use case.

Maybe there is a smart way to add a lot of keys. But it would require more cmake code.
Maybe add a COCOA_KEYS input, that expect a list of key.
`COCOA_KEYS ${MY_KEY_1} ${MY_KEY_2}`

Key could be created like that:
```
set(MY_KEY_1 MyKey1)
set(${MY_KEY_1}_TYPE "string")
set(${MY_KEY_1}_KEY "NSPhotoLibraryUsageDescription")
set(${MY_KEY_1}_DESCRIPTION "My Description")
```

Which could be wrapped into a `function qt_ios_create_string_key`, and then also do a `function qt_ios_create_bool_key`, etc... for every keys.

Then in QtIosCMake simply loop on each key and append into a string EXTRA_KEYS that will go in Info.plist.in.

This is doable quite easily, but i need a project with already a bunch of keys to tests/debug/compare result and see what is expected.

I haven't used lots of cocoa keys yet.

Anyway for now i can just merge your PR that is common key, i will see later to add something smart.

**Review by OlivierLDff — APPROVED** [review]

**Comment by OlivierLDff** [comment]
Thanks for the PR !

**Pull request merged**